### PR TITLE
Gdr 2214

### PR DESCRIPTION
**Pull request by bczech**
# Description
## What changed?
Related JIRA issue: GDR-2214 

## Why was it changed?
To properly visualize and present averaged data in gDRsearch2

# Checklist for sustainable code base
- [ ] I added tests for any code changed/added
- [ ] I added documentation for any code changed/added
- [ ] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [X] Package version bumped
- [X] Changelog updated

# Screenshots (optional)

### CODE REVIEW

**Review by marcin-kam — APPROVED** [review]

File: R/flatten.R [hunk 1]
```diff
@@ -58,11 +58,17 @@ flatten <- function(tbl, groups, wide_cols, sep = "_") {
...
+        paste0(grp, sep, colnames(group)[rename])
+      }
     out[[grp]] <- group
   }
 
   ## Drop empty elements for successful merge.
   filtered <- out[lapply(out, nrow) > 0L]
-  Reduce(function(x, y) x[y, on = intersect(names(x), names(y))], filtered)
+  Reduce(function(x, y) x[y, on = intersect(names(x), names(y)),
+                          allow.cartesian = TRUE], filtered)
```
Comment (j-smola) [inline R/flatten.R:73]: 1. lintr said or 
```
Reduce(function(x, y) x[y, on = intersect(names(x), names(y)), allow.cartesian = TRUE], filtered)
```
or use  `function(x,y) {...}`
2. Are we sure we will allow cartesian?

File: R/flatten.R [hunk 2]
```diff
@@ -58,11 +58,17 @@ flatten <- function(tbl, groups, wide_cols, sep = "_") {
   rename <- colnames(out[[1]]) %in% wide_cols
   for (grp in names(out)) {
     group <- out[[grp]]
-    colnames(group)[rename] <- paste0(grp, sep, colnames(group)[rename])
+    colnames(group)[rename] <- if (any("x" == colnames(group))) {
```
Comment (j-smola) [inline R/flatten.R:61]: Just from logical point of view - maybe would be better `any(colnames(group)) == "x")` ?

**Review by j-smola — COMMENTED** [review]
Could you maybe add test for data with columns "x" and "x_std"?

File: R/flatten.R (hunk 2 above)
Comment (bczech) [reply R/flatten.R]: Done

**Pull request merged**